File: languages/elixir/gladiator.exs
s = IO.read(:stdio, :line)
out = ""
if String.contains?(s, "\"discipline\":\"construct\"") do
  out = if String.contains?(s, "\"arena\":\"cheeks\""), do: "()", else: out
  out = if String.contains?(s, "\"arena\":\"ears\""), do: "/\\\\/\\\\", else: out
  out = if String.contains?(s, "\"arena\":\"eyes\""), do: "oo", else: out
  out = if String.contains?(s, "\"arena\":\"geometry\""), do: "", else: out
  out = if String.contains?(s, "\"arena\":\"head_top\""), do: "_", else: out
  out = if String.contains?(s, "\"arena\":\"mouth\""), do: "^", else: out
  out = if String.contains?(s, "\"arena\":\"newlines\""), do: "", else: out
  out = if String.contains?(s, "\"arena\":\"nose\""), do: ".", else: out
  out = if String.contains?(s, "\"arena\":\"padding\""), do: "      ", else: out
  out = if String.contains?(s, "\"arena\":\"whiskers\""), do: "><", else: out
  IO.write("{\"ok\":true,\"output\":\"" <> out <> "\"}")
else
  ok = String.contains?(s, "\"candidate\":\" /\\\\_/\\\\\\n( o.o )\\n > ^ <\"")
  v = if ok, do: "es un gato", else: "no es el gato canonico"
  bo = if ok, do: "true", else: "false"
  IO.write("{\"ok\":" <> bo <> ",\"output\":\"" <> v <> "\"}")
end

File: languages/lua/gladiator.lua
local s = io.read('*a')
local out = ""
if s:find("\"discipline\":\"construct\"", 1, true) then
  if s:find("\"arena\":\"cheeks\"", 1, true) then out = "()"
  elseif s:find("\"arena\":\"ears\"", 1, true) then out = "/\\\\/\\\\"
  elseif s:find("\"arena\":\"eyes\"", 1, true) then out = "oo"
  elseif s:find("\"arena\":\"geometry\"", 1, true) then out = ""
  elseif s:find("\"arena\":\"head_top\"", 1, true) then out = "_"
  elseif s:find("\"arena\":\"mouth\"", 1, true) then out = "^"
  elseif s:find("\"arena\":\"newlines\"", 1, true) then out = ""
  elseif s:find("\"arena\":\"nose\"", 1, true) then out = "."
  elseif s:find("\"arena\":\"padding\"", 1, true) then out = "      "
  elseif s:find("\"arena\":\"whiskers\"", 1, true) then out = "><"
  end
  io.write('{"ok":true,"output":"' .. out .. '"}')
else
  if s:find("\"candidate\":\" /\\\\_/\\\\\\n( o.o )\\n > ^ <\"", 1, true) then
    io.write('{"ok":true,"output":"' .. "es un gato" .. '"}')
  else
    io.write('{"ok":false,"output":"' .. "no es el gato canonico" .. '"}')
  end
end
